docs/protocol_overview.md

```diff
@@ -0,0 +1,245 @@
+
+### Rarely-online nodes
+
+One of the most important conditions for 'simple' and 'lightweight' node operations is how often the node needs to be online.
+If we want to make the validators run nodes on their laptops, it is not realistic to assume that they will be online 24/7.
+
+Since Simperby's BFT consensus assumes a partially-synchronous network, rarely-online nodes can be trivially handled because it's no more than one typical case of an asynchrony, if
+
+1. the consensus round is (or has grown to be) long enough to cover all appearances of the nodes.
+2. at least one node is online when there is a network broadcast
+
+Condition 1 turns out to be a tough challenge in the later sections, so keep it in mind.
+
+### On-demand block production
+
+Rarely-online nodes mean rarely-progressed blockchain.
+(note that in any BFT algorithm, at least 2/3 of nodes must have participated in the consensus process to produce a block)
+
+This will inevitably compromise the latency or throughput of the blockchain, but it doesn't matter because
+**Simperby is NOT a general contract platform for serving public users,**
+**but a governance platform for a set of permissioned members**.
+
+Simperby blockchain includes only governance-approved state transitions (there are very [few exceptions](#consensus-leader)).
+Every governance agenda must be approved by a majority vote of the members so it will take time(days or even weeks).
+Considering that, naturally, Simperby's performance is bound by the governance process, not the consensus which is slowed by lightweight node operation.
+
+### Multichain DAO
+
+One of the notable use cases of Simperby is that it can establish a **muli-chain DAO**. (See [this](./multichain_dao.md) for more details)
+It's super-easy to expand the organization to other existing chains (so-called 'colony chains') using
+the light client of the Simperby consensus.
+Since the light client is uploaded as a contract to other chains,
+any block progress of the Simperby consensus will require an additional
+transaction to update and store the Merkle root of it.
+This might not be cheap, because **the organization will have to pay for the gas cost for every colony chain, for every Simperby block**.
+That is another reason that on-demand block production is considered reasonable.
+
+## Governance
+
+Simperby's governance is implemented by P2P voting.
+
+- Any member can propose an agenda, and it will be propagated over the P2P network.
+- Any member also can cast a vote on the agenda if they prefer it. The vote will be propagated as well.
+- Any agenda that has over 50% votes is **eligible** to be included in a block.
+- A block can be **valid only if it includes an eligible agenda**. In other words, empty blocks are not allowed.
+- To decide which eligible agenda to finalize is the role of consensus, not the governance.
+- The consensus is a BFT algorithm based on a leader-and-round style.
+
+### What can be an agenda?
```

Comment:
How about adding a description of the voting process in this section?
(a sentence like "A transaction includes a human-readable description of it, and each individual reads the description to determine whether to cast a vote or not.")

Reply:
Actually, it's not that simple. (especially after I introduce the transaction-as-a-git-commit protocol)
The definitions of 'transaction', 'state', and 'state transition' are beyond the scope of this documents.
Focusing only on the governance and consensus, leaving the higher layer abstract will make this document more concise.

Reply:
Okay. I agree with you.